client/packages/common/src/ui/components/inputs/DatePickers/BaseDatePickerInput/BaseDatePickerInput.tsx

```diff
@@ -1,16 +1,42 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import {
   DatePicker,
   DatePickerProps,
 } from '@mui/x-date-pickers/DatePicker/DatePicker';
 import { BasicTextInput } from '../../TextInput/BasicTextInput';
 import { StandardTextFieldProps, TextFieldProps } from '@mui/material';
 import { useAppTheme } from '@common/styles';
+import { DateUtils } from '@common/intl';
+import { useDebounceCallback } from '@common/hooks';
 
 export const BaseDatePickerInput: FC<
-  Omit<DatePickerProps<Date, Date>, 'renderInput'>
+  Omit<DatePickerProps<Date, Date>, 'renderInput' | 'value'> & {
+    value: Date | string | null;
```

Comment:
The passed in `value` can be either a date or a string (different implementations do it differently).